MAE-220: Fix Issues with DD E-mail Templates

MiyaNoctem opened the pull request:
## Overview
Several fixes needed to be done on the DD e-mail templates:

1. Membership Amount shown on DD email notification is not included with VAT value. If you check the contribution amount & total amount, VAT is not included, while in some places of the email, VAT is included in the contribution amount.
2. Direct Debit Logo is missing on the email.
3. Spacing of the Direct Debit Guarantee block was too large.
4. Dates should be in CiviCRM short date format, as per configured settings on all the instances.

## Before
1. Tax was not taken into account when building the templates.
2. The URL for the logo was being built as a relative URL, but on the e-mail we need the absolute URL to be able to locate the image.
3. 60px were hard-coded as spacing before the guarantee block.
4. Dates were being shown in yyyy-mm-dd format.

![image](https://user-images.githubusercontent.com/21999940/79336300-18fdfe00-7ee9-11ea-9e81-17983cc804fe.png)

## After
1. Added taxes to records and totals, to be used in templates, and showed tax values where applicable, if contributions actually have tax.
2. Built the absolute URL so the image gets resolved when opening the e-mail.
2. Reduced spacing to 30px.
4. Obtained CiviCRM shortdate setting and passed format to all the templates, so it can be used to print dates in this format.

![image](https://user-images.githubusercontent.com/21999940/79336123-cae8fa80-7ee8-11ea-977e-f92ace9f54ec.png)

jamienovick commented:
@MiyaNoctem re:
2. The URL for the logo was being built as a relative URL, but on the e-mail we need the absolute URL to be able to locate the image.

Where is the URL stored? What if it moved on the server?

4. Should these not respect the date formats that CiviCRM uses and therefor be a configuration and not hard coded?

MiyaNoctem commented:
@jamienovick , @omarabuhussein , I've added the fix for MAE-221, done on #156 and also updated the wording so it now reads "To be payed in x installments of y". I will be closing the PR for MAE-221 in favor of this one with all the fixes.

cc @ellencompucorp

omarabuhussein commented on CRM/ManualDirectDebit/Mail/DataCollector/Base.php:
```diff
@@ -220,7 +220,7 @@ private function collectRecurringContributionData() {
     $recurringContributionRows['recurringInstallmentsTable'] = $this->buildRecuringContributionTable($recurringContributionPlan, $totalTax);
     $total = $this->formatAmount($total);
-    $totalTax = $this->formatAmount($totalTax);
+    $totalTax = $totalTax ? $this->formatAmount($totalTax) : null;
```
we use Capital letter NULL

omarabuhussein commented on CRM/ManualDirectDebit/Mail/DataCollector/Base.php:
```diff
@@ -351,7 +351,7 @@ private function collectOrderLineItems() {
         'entityTable' => $lineItem['entity_table'],
         'entityId' => $lineItem['entity_id'],
-        'tax' => $this->formatAmount($tax),
+        'tax' => $tax ? $this->formatAmount($tax) : null,
```
same here
